client/quantum_serverless/core/job.py

```diff
@@ -455,54 +455,22 @@ def upload(self, program: QiskitFunction):
         tracer = trace.get_tracer("client.tracer")
         with tracer.start_as_current_span("job.run") as span:
             span.set_attribute("program", program.title)
-
             url = f"{self.host}/api/{self.version}/programs/upload/"
-            artifact_file_path = os.path.join(program.working_dir, "artifact.tar")
 
-            # check if entrypoint exists
-            if not os.path.exists(
-                os.path.join(program.working_dir, program.entrypoint)
-            ):
-                raise QuantumServerlessException(
-                    f"Entrypoint file [{program.entrypoint}] does not exist "
-                    f"in [{program.working_dir}] working directory."
+            if program.image is not None:
+                # upload function with custom image
+                program_title = _upload_with_docker_image(
+                    program=program, url=url, token=self._token, span=span
                 )
-
-            with tarfile.open(artifact_file_path, "w") as tar:
-                for filename in os.listdir(program.working_dir):
-                    fpath = os.path.join(program.working_dir, filename)
-                    tar.add(fpath, arcname=filename)
-
-            # check file size
-            size_in_mb = Path(artifact_file_path).stat().st_size / 1024**2
-            if size_in_mb > MAX_ARTIFACT_FILE_SIZE_MB:
-                raise QuantumServerlessException(
-                    f"{artifact_file_path} is {int(size_in_mb)} Mb, "
-                    f"which is greater than {MAX_ARTIFACT_FILE_SIZE_MB} allowed. "
-                    f"Try to reduce size of `working_dir`."
+            elif program.entrypoint is not None:
```

Comment:
Is it good to put a waring message when both image and entrypoint are set?

Reply:
I think it would be better to move all validation to QiskitFunction itself to reduce boilderplate code. Created an iissue for that. https://github.com/Qiskit-Extensions/qiskit-serverless/issues/1315

Thanks for suggestion!